### ISSUE TITLE
[BUG] Generating support bundle may get stuck

### ISSUE THREAD

**Issue by shuo-wu** [longhorn/longhorn#4342]
## Describe the bug

After the discussion of [the comment](https://github.com/longhorn/longhorn/issues/4255#issuecomment-1201826741), the following ideas may help fix the issue:
- [ ] Have a timeout mechanism to handle the stuck streaming during the support bundle generation. After the timeout, the log collection should continue.
- [ ] Inform users of the failure after the timeout, so that they can decide if they need to handle the collection of the missing log manually.
- [ ] Or fall back to fetch the most recent logs if there is a timeout

## To Reproduce

Steps to reproduce the behavior (Unverified):
1. Make some of the Longhorn workloads hold lots of logs. (e.g., Disable the scheduling then launch lots of new replicas, then longhorn managers would contain tons of scheduling error logs.)
2. Generate the support bundle via UI

## Expected behavior
The generation should complete or timeout with details

## Environment

 - Longhorn version: v1.3.x or v1.2.x

## Additional context
N/A

**Comment by innobead** [comment]
@c3y1huang is this still an issue for the new support bundle mechanism after 1.4 included?

**Comment by innobead** [comment]
cc @longhorn/qa

**Comment by c3y1huang** [comment]
> @c3y1huang is this still an issue for the new support bundle mechanism after 1.4 included?
> 
> #6256

Yes, I think it's possible that this issue still exists in the support-bundle-kit. 
- https://github.com/rancher/support-bundle-kit/blob/v0.0.25/pkg/manager/cluster.go#L162
- https://github.com/rancher/support-bundle-kit/blob/v0.0.25/pkg/manager/manager.go#L169

**Comment by samip5** [comment]
It indeed still is somewhat broken for example if one node is for reasons unable to send it's bundle to the longhorn backend, it will get stuck.
